go-runtime/compile/schema.go

```diff
@@ -621,13 +629,13 @@ func visitStruct(pctx *parseContext, pos token.Pos, tnode types.Type) (*schema.R
 		var metadata []schema.Metadata
 		if jsonFieldName != "" {
 			metadata = append(metadata, &schema.MetadataAlias{
-				Pos:   goPosToSchemaPos(pos),
+				Pos:   goPosToSchemaPos(f.Pos()),
 				Kind:  schema.AliasKindJSON,
 				Alias: jsonFieldName,
 			})
 		}
 		out.Fields = append(out.Fields, &schema.Field{
-			Pos:      goPosToSchemaPos(pos),
+			Pos:      goPosToSchemaPos(f.Pos()),
```

Comment:
I think these all needed to be `f.Pos()` since we're validating the `s.Field(i)` here.